scripts/webpack/plugins/HTMLWebpackCSSChunks.js

```diff
@@ -0,0 +1,42 @@
+const HtmlWebpackPlugin = require('html-webpack-plugin');
+
+/*
+ * This plugin returns the css associated with entrypoints. Those chunks can be found
+ * in `htmlWebpackPlugin.files.cssChunks`.
+ * The HTML Webpack plugin removed the chunks object in v5 in favour of an array however if we want
+ * to do anything smart with hashing (e.g. [contenthash]) we need a map of { themeName: chunkNameWithHash }.
+ */
+class HTMLWebpackCSSChunks {
+  /**
+   * @param {import('webpack').Compiler} compiler
+   */
+  apply(compiler) {
+    compiler.hooks.compilation.tap(
+      'HTMLWebpackCSSChunks',
+      /**
+       * @param {import('webpack').Compilation} compilation
+       */
+      (compilation) => {
+        HtmlWebpackPlugin.getHooks(compilation).beforeAssetTagGeneration.tapAsync(
+          'HTMLWebpackCSSChunks',
+          (data, cb) => {
+            data.assets.cssChunks = {};
+
+            for (const entryPoint of compilation.entrypoints.values()) {
+              for (const chunk of entryPoint.chunks) {
+                const cssFile = [...chunk.files].find((file) => file.endsWith('.css'));
```

Comment:
**Question:** maybe a dumb question, but what if there are multiple chunk files ending with `.css`?

Reply:
Not a dumb question at all. This plugin is very geared towards our current usage of defining the css theme files as entry points which only ever output a single `css` file per entry (`dark`, `light`). Given we are aiming to migrate styles to emotion and remove scss from the codebase I wrote this as "dumb" as necessary to get the job done.

The alternative I considered to writing this plugin was to change the scss to output a single css file that used css custom variables for theming but felt that would introduce A LOT of changes and potentially styling bugs.